drivers/sensor/bmi160/bmi160.c

```diff
@@ -712,20 +715,33 @@ static int bmi160_sample_fetch(const struct device *dev,
 	struct bmi160_data *data = dev->data;
 	uint8_t status;
 	size_t i;
+	int ret = 0;
+	enum pm_device_state pm_state;
+
+	pm_device_busy_set(dev);
```

Comment:
That is only used if you build without `PM_DEVICE_RUNTIME_EXCLUSIVE` that is enabled by default when `PM_DEVICE_RUNTIME` is selected. That's said it is reasonable to have it if you don't want the device suspending when the system is idle and you were built without device runtime support.

Reply:
This is one of those devices that probably should not be turned off unless explicitly told to do so as it takes so long to come back from suspend (100's of ms).

So right, what I think I want is to not suspend this device if triggers are set or samples are being fetched. I need to add the busy set on trigger set as well, and clear it if the func is NULL.

I need to read more on the system idle part of PM I think to make sure that's what I'd want though.

Reply:
If you are exclusive using device runtime power management you don't need the busy flag, it is enough to hold a device reference (`pm_device_runtime_get`). The busy flag is handy if this device can be built without device runtime power management and you don't want the device be suspended when the soc is idle.

Reply:
If that's the case I probably need some conditional inclusion of power management for the device based on whether or not PM_DEVICE_RUNTIME is enabled?

I think that's probably fine, so no PM_DEVICE work is done unless runtime power management is enabled. As in don't do anything unless the application is in charge of PM.

I want to make sure I get this right so I can replicate the pattern and document it correctly